### ISSUE TITLE
Prefix doesn't apply when using fieldName

### ISSUE THREAD

**Issue by parisholley** [mikro-orm/mikro-orm#2361]
```typescript
@Embeddable()
export class Nested {
  @Property({ fieldName: 'foobar' })
  child: string;
}

@Entity()
export default class MyModel {
  @Embedded(() => Nested, {prefix: 'nested_'})
  nested: Nested;
}
```

I would expect the above to produce `nested_foobar` in queries, but it just uses `foobar`. if i remove the fieldName, it will just fallback to the using the library naming strategy vs being overriden. i'm hitting this because of production schema produced by another ORM

```
    "@mikro-orm/cli": "^4.5.9",
    "@mikro-orm/core": "^4.5.9",
    "@mikro-orm/migrations": "^4.5.9",
    "@mikro-orm/nestjs": "^4.3.1",
    "@mikro-orm/postgresql": "^4.5.9",
    "@mikro-orm/reflection": "^4.5.9",
```

**Comment by derevnjuk** [comment]
Any updates? Why don't easily extend the naming strategy by adding a method to retrieve a column name for embedded fields?

**Comment by B4nan** [comment]
This is unfortunately much more complicated. It requires changes on hydration level and they would be breaking, or we would have to have the hydration for such happening twice, as we still want to support the property name too.

**Comment by SPHeWa** [comment]
Are there any Updates to this Issue? I am facing the same problem.

**Comment by B4nan** [comment]
Closing as fixed in v6 via #4866
